pkg/resources/validators_test.go

```diff
@@ -0,0 +1,189 @@
+			CheckingFn: accountObjectIdentifierCheck,
+		},
+		{
+			Name:       "validation: incorrect form for account object identifier",
+			Value:      "a.b",
+			Error:      "<name>, but was <database_name>.<name>",
+			CheckingFn: accountObjectIdentifierCheck,
+		},
+		{
+			Name:       "validation: incorrect form for database object identifier",
+			Value:      "a.b.c",
+			Error:      "<database_name>.<name>, but was <database_name>.<schema_name>.<name>",
+			CheckingFn: databaseObjectIdentifierCheck,
+		},
+		{
+			Name:       "validation: incorrect form for schema object identifier",
+			Value:      "a.b.c.d",
+			Error:      "<database_name>.<schema_name>.<name>, but was <database_name>.<schema_name>.<table_name>.<column_name>",
+			CheckingFn: schemaObjectIdentifierCheck,
+		},
+		{
+			Name:       "validation: incorrect form for table column identifier",
+			Value:      "a",
+			Error:      "<database_name>.<schema_name>.<table_name>.<column_name>, but was <name>",
+			CheckingFn: tableColumnIdentifierCheck,
+		},
+		{
+			Name:       "correct form for account object identifier",
+			Value:      "a",
+			CheckingFn: accountObjectIdentifierCheck,
+		},
+		{
+			Name:       "correct form for database object identifier",
+			Value:      "a.b",
+			CheckingFn: databaseObjectIdentifierCheck,
+		},
+		{
+			Name:       "correct form for schema object identifier",
+			Value:      "a.b.c",
+			CheckingFn: schemaObjectIdentifierCheck,
+		},
+		{
+			Name:       "correct form for table column identifier",
+			Value:      "a.b.c.d",
+			CheckingFn: tableColumnIdentifierCheck,
+		},
+	}
+
+	for _, tt := range testCases {
+		t.Run(tt.Name, func(t *testing.T) {
+			diag := tt.CheckingFn(tt.Value, cty.IndexStringPath("path"))
+			if tt.Error != "" {
+				assert.Len(t, diag, 1)
+				assert.Contains(t, diag[0].Detail, tt.Error)
+			} else {
+				assert.Len(t, diag, 0)
+			}
+		})
+	}
+}
+
+func TestGetExpectedIdentifierForm(t *testing.T) {
+	testCases := []struct {
+		Name     string
+		Expected string
+		Actual   string
+	}{
+		{
+			Name:     "correct account object identifier from generic parameter",
+			Expected: "<name>",
+			Actual:   getExpectedIdentifierRepresentationFromGeneric[sdk.AccountObjectIdentifier](),
+		},
+		{
+			Name:     "correct database object identifier from generic parameter",
+			Expected: "<database_name>.<name>",
+			Actual:   getExpectedIdentifierRepresentationFromGeneric[sdk.DatabaseObjectIdentifier](),
+		},
+		{
+			Name:     "correct schema object identifier from generic parameter",
+			Expected: "<database_name>.<schema_name>.<name>",
+			Actual:   getExpectedIdentifierRepresentationFromGeneric[sdk.SchemaObjectIdentifier](),
+		},
+		{
+			Name:     "correct table column identifier from generic parameter",
+			Expected: "<database_name>.<schema_name>.<table_name>.<column_name>",
+			Actual:   getExpectedIdentifierRepresentationFromGeneric[sdk.TableColumnIdentifier](),
+		},
+		{
```

Comment:
It would be nice to split these 4 into separate one (then we can get rid of getExpectedIdentifierRepresentationFromParam in testcase def - move it to body). ALso in body we can then test two options: ptr and no ptr one.

Reply:
Done